v0/destinations/kafka/transform.js

```diff
@@ -18,19 +19,39 @@ function batch(destEvents) {
 }
 
 function process(event) {
-  const integrationsObj = getIntegrationsObj(event.message, "kafka");
-  const { schemaId } = integrationsObj || {};
+  const { message } = event;
+  const integrationsObj = getIntegrationsObj(message, "kafka");
+  const { schemaId, topic } = integrationsObj || {};
+  const userId = message.userId || message.anonymousId;
   if (schemaId) {
     return {
-      message: event.message,
-      userId: event.message.userId || event.message.anonymousId,
-      schemaId
+      message,
+      userId,
+      schemaId,
+      topic
     };
   }
   return {
-    message: event.message,
-    userId: event.message.userId || event.message.anonymousId
+    message,
+    userId,
+    topic
   };
 }
 
-module.exports = { process, batch };
+/**
+ * This functions takes event matadata and updates it based on the transformed and raw paylaod
+ * @param {*} input
+ * @returns {*} metadata
+ */
+function processMetadata(input) {
+  const { metadata, inputEvent } = input;
+  const clonedMetadata = cloneDeep(metadata);
+  const integrationsObj = getIntegrationsObj(inputEvent.message, "kafka");
+  const { topic } = integrationsObj || {};
+  if (topic) {
+    clonedMetadata.rudderId = `${clonedMetadata.rudderId}<<>>${topic}`;
```

Comment:
`<<>>` is there any significance to this separator?

Reply:
its for keeping parity no significant meaning
original rudderId is stored in this format `anonymousId<<>>userId`